app/lib/homework/homework_dialog/homework_dialog_bloc.dart

```diff
@@ -792,13 +794,17 @@ class HomeworkDialogBloc extends Bloc<HomeworkDialogEvent, HomeworkDialogState>
   }
 
   Date _getNextSchoolday() {
-    final today = _clock.now().toDate();
-    final daysUntilNextSchoolday = switch (today.weekDayEnum) {
-      WeekDay.friday => 3, // Monday
-      WeekDay.saturday => 2, // Monday
-      _ => 1 // Tomorrow
-    };
-    return today.addDays(daysUntilNextSchoolday);
+    var candidate = _clock.now().toDate();
+    // hope this code is refactored by then 🤡
+    while (candidate.year < 2050) {
+      candidate = candidate.addDays(1);
+      if (enabledWeekdays.contains(candidate.weekDayEnum)) {
+        return candidate;
+      }
+    }
+
+    // Should never happen, but who knows ¯\_(ツ)_/¯
+    return _clock.now().toDate().addDays(1);
```

Comment:
This is already handled by the `while (candidate.year < 2050)`, i.e. if it is empty the return at the very bottom will just return tomorrow. It doesn't matter that it is more computation because this is an edge case that no normal user will ever hit. I didn't want to use `while(true)` as the current logic will at least have an "emergency exit" condition if we f*** up some logic e.g. in `candidate.addDays(1)` in the future.